tests/src/test/scala/com/nvidia/spark/rapids/unit/StringUnitTest.scala

```diff
@@ -0,0 +1,159 @@
+            withResource(result.copyToHost()) { hostResult =>
+              assertResult(hostResult.getRowCount)(1)
+              assertResult(hostResult.isNull(0))(true)
+            }
+          }
+        }
+
+      doTest(-1)
+      doTest(0)
+      doTest(1)
+    }
+
+    // Test repeat(str, intVal).
+    withResource(GpuScalar("abc123", DataTypes.StringType)) { strScalar =>
+      val doTest = (intVal: Int, expectedStr: String) =>
+        withResource(GpuScalar(intVal, DataTypes.IntegerType)) { intScalar =>
+          withResource(GpuStringRepeat(null, null).doColumnar(1, strScalar, intScalar)) {
+            result =>
+              withResource(result.copyToHost()) { hostResult =>
+                assertResult(hostResult.getRowCount)(1)
+                assertResult(hostResult.isNull(0))(false)
+                assertResult(hostResult.getJavaString(0))(expectedStr)
+              }
+          }
+        }
+
+      doTest(-1, "")
+      doTest(0, "")
+      doTest(1, "abc123")
+      doTest(2, "abc123abc123")
+      doTest(3, "abc123abc123abc123")
+    }
+  }
+
+  test("Test StringRepeat with scalar string and column repeatTimes") {
+    val intCol = ColumnVector.fromBoxedInts(-3, null, -1, 0, 1, 2, null)
+
+    withResource(GpuColumnVector.from(intCol, DataTypes.IntegerType)) { intGpuColumn =>
+      // Test repeat(NULL, intVal).
+      withResource(GpuScalar(null, DataTypes.StringType)) { nullStr =>
+        withResource(GpuStringRepeat(null, null).doColumnar(nullStr, intGpuColumn)) { result =>
+          withResource(result.copyToHost()) { hostResult =>
+            (0 until hostResult.getRowCount.asInstanceOf[Int]).foreach { i =>
+              assertResult(hostResult.isNull(0))(true)
+            }
+          }
+        }
+      }
+
+      // Test repeat(str, intVal).
+      withResource(GpuScalar("abc123", DataTypes.StringType)) { strScalar =>
+        withResource(GpuStringRepeat(null, null).doColumnar(strScalar, intGpuColumn)) { result =>
+          withResource(result.copyToHost()) { hostResult =>
+            CudfTestHelper.assertColumnsAreEqual(
+              HostColumnVector.fromStrings("", null, "", "", "abc123", "abc123abc123", null),
+              hostResult,
+              "Repeated strings")
+          }
+        }
+      }
+    }
+  }
+
+  test("Test StringRepeat with column strings and scalar repeatTimes") {
+    val strsCol = ColumnVector.fromStrings(null, "a", "", "123", "éá")
+
+    withResource(GpuColumnVector.from(strsCol, DataTypes.StringType)) { strGpuColumn =>
+      // Test repeat(strs, NULL).
+      withResource(GpuScalar(null, DataTypes.IntegerType)) { nullInt =>
+        withResource(GpuStringRepeat(null, null).doColumnar(strGpuColumn, nullInt)) { result =>
+          withResource(result.copyToHost()) { hostResult =>
+            (0 until hostResult.getRowCount.asInstanceOf[Int]).foreach { i =>
+              assertResult(hostResult.isNull(0))(true)
```

Comment:
This is not using `i`, only checking the first row `i` times.

Could simplify this and avoid the host transfer with something like:
```
  assert(strGpuColumn.getRowCount)(result.getRowCount)
  assertResult(result.getNullCount)(result.getRowCount)
```